lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -968,10 +874,30 @@ def __get_stored_workflow(self, trans, workflow_id, **kwd):
     description="Set this to true to skip joining workflow step counts and optimize the resulting index query. Response objects will not contain step counts.",
 )
 
+InvokeWorkflowBody = Annotated[
+    InvokeWorkflowPayload,
+    Body(
+        default=...,
+        title="Invoke workflow",
+        description="The values to invoke a workflow.",
+    ),
+]
+
+RefactorWorkflowBody = Annotated[
+    RefactorRequest,
+    Body(
+        default=...,
+        title="Refactor workflow",
+        description="The values to refactor a workflow.",
+    ),
+]
+
 
 @router.cbv
 class FastAPIWorkflows:
     service: WorkflowsService = depends(WorkflowsService)
+    manager: WorkflowsManager = depends(WorkflowsManager)
+    contents_manager: WorkflowContentsManager = depends(WorkflowContentsManager)
```

Comment:
Your WorkflowService has already an instance of WorkflowsManager and WorkflowContentsManager in case you want to move the logic using these 2 to the service.